src/components/data/data-utils.js

```diff
@@ -438,14 +440,14 @@ export const EntityUploader = ({ onSuccess, onDismiss, namespace, name, entityTy
             }
           })
         ]),
-        currentFile && _.includes(_.toLower(newEntityType), entityTypes) && div({
+        ((currentFile && entityAlreadyExists) || recordAlreadyExists) && div({
           style: { ...warningBoxStyle, margin: '1rem 0 0.5rem', display: 'flex', alignItems: 'center' }
         }, [
           icon('warning-standard', { size: 19, style: { color: colors.warning(), flex: 'none', marginRight: '0.5rem', marginLeft: '-0.5rem' } }),
-          `Data with the type '${newEntityType}' already exists in this workspace. `,
+          `Data with the type '${recordType ? recordType : newEntityType}' already exists in this workspace. `,
           'Uploading more data for the same type may overwrite some entries.'
         ]),
-        currentFile && containsNullValues && _.includes(_.toLower(newEntityType), entityTypes) && div({
+        currentFile && containsNullValues && (entityAlreadyExists || recordAlreadyExists) && div({
```

Comment:
I don't think WDS supports the "We have detected empty cells in your TSV. Please choose an option:" feature. Could you remove that part of the dialog for WDS?